### ISSUE TITLE
Eclipse hard shadows appear to be off by a few KM.

### ISSUE THREAD

**Issue by micahnyc** [OpenSpace/OpenSpace#1482]
As seen in attached images in comparison to eclipse tracker websites showing path of totality. 

![image](https://user-images.githubusercontent.com/708535/106507759-7f183980-6499-11eb-9d25-7b039bbd3482.png)

![image](https://user-images.githubusercontent.com/708535/106508343-43ca3a80-649a-11eb-9f80-c1a137dd8dcc.png)

https://eclipse.gsfc.nasa.gov/SEgoogle/SEgoogle2001/SE2020Dec14Tgoogle.html
http://xjubier.free.fr/en/site_pages/solar_eclipses/TSE_2017_GoogleMapFull.html

**Comment by micahnyc** [comment]
(As reported by Ken at SMSU)

**Comment by hedbergj** [comment]
Maybe the perfectly spherical earth?

```
 --local earthEllipsoid = { 6378137.0, 6378137.0, 6356752.314245 }
local earthEllipsoid = { 6378137.0, 6378137.0, 6378137.0 }
```

**Comment by alexanderbock** [comment]
Good point!

@jccosta   What was the reason for the non tri-axial ellipsoid?  Shouldn't it be possible to apply a non uniform scaling to the scene graph now that we have that type?

**Comment by jccosta** [comment]
@alexanderbock, the reason for the non-tri-axial ellipsoid is partly because of the eclipse shadow's calculation and partly because of the atmosphere's calculations.
I think the problem with the eclipses' shadows alignment is one of (or all) the following (order by what I think is more probable):
1) The initial premises when calculating the eclipses. It is a geometric approximation of planets and the Sun in 2D, extended to 3D. Here, the Sun and the planets are considered perfect spheres
2) An error in the eclipses' calculation
3) Precision issues in the calculations (we are using double precision for all calculations, but we are talking about comparisons between large distances and small position on the globe)
4) GDAL tile positioning (although I think this is very improbable of happening, if the three points above are discarded, we may need to investigate this further)

**Comment by hedbergj** [comment]
Certainly hits St. Louis better at 18:19 on Aug 21, 2017, when the earth is oblate:

``` local earthEllipsoid = { 6378137.0, 6378137.0, 6356752.314245 }```

(Though I'm not sure why my shadows are so chunky? - different issue I guess)

![OpenSpace_000023](https://user-images.githubusercontent.com/7926907/106624976-38cbe480-6544-11eb-8ca2-3cc7161ddc66.jpg)

**Comment by alexanderbock** [comment]
To check:   Maybe it is related to the SSB/Sun position offset that we had before

**Comment by WeirdRubberDuck** [comment]
So, the position of the eclipse was not improved by the SSB/Sun position change (image from master as of July 5, ffd3514b773db823b644f582a2d8fca8aee3740f) 
![image](https://user-images.githubusercontent.com/8808894/124751872-6bc93880-df27-11eb-85ca-1d389f511824.png)

**Comment by hedbergj** [comment]
with the spheroid radii in v 17

` local earthEllipsoid = { 6378137.0, 6378137.0, 6356752.314245 }
`

![eclipse-stlouis](https://user-images.githubusercontent.com/7926907/124794801-649d2d00-df1d-11eb-8477-e49cf3ede0b5.jpg)

Though, things might now be off by a minute or two (that I don't remember from before). 
at 18:19 GMT:

![OpenSpace_000020](https://user-images.githubusercontent.com/7926907/124795303-e1c8a200-df1d-11eb-8237-2a672b6e9d52.png)

vs.

![google-eclipse](https://user-images.githubusercontent.com/7926907/124795435-0886d880-df1e-11eb-84e9-7227739f732b.jpg)

**Comment by hedbergj** [comment]
Also, this was done with the oblate spheroid for the Earth asset. (and some tweaks to atmosphere so that the height at NYC latitude was correct.) It seemed pretty close to what others predicted. 

![eclipse-stages-2-annotated](https://user-images.githubusercontent.com/7926907/124796391-1b4ddd00-df1f-11eb-9897-3fb89451afec.png)

**Comment by WeirdRubberDuck** [comment]
> Though, things might now be off by a minute or two (that I don't remember from before).

Good catch @hedbergj. My first thought is that maybe the shadow might be offset with the same difference that the sun position had after the SSB/Sun position update... 🤔 Someone gotta look into this

**Comment by hedbergj** [comment]
Yeah @WeirdRubberDuck - that's probably the case. My current master is from before the [atm updates](https://github.com/OpenSpace/OpenSpace/commit/20aa307ece1bfbff1e79de76c2a4b9ac89cf389e) @alexanderbock did very recently. Those might fix the timing for the atmosphere shadows, but then also the globebrowsing tile eclipse shadows might need to be looked at too. Changes in SSB/Sun should affect longitudinal (e/w) errors more than latitude (n/s) I bet.

**Comment by WeirdRubberDuck** [comment]
> Yeah @WeirdRubberDuck - that's probably the case. My current master is from before the [atm updates](https://github.com/OpenSpace/OpenSpace/commit/20aa307ece1bfbff1e79de76c2a4b9ac89cf389e) @alexanderbock did very recently. Those might fix the timing for the atmosphere shadows, but then also the globebrowsing tile eclipse shadows might need to be looked at too. Changes in SSB/Sun should affect longitudinal (e/w) errors more than latitude (n/s) I bet.

Ah, I see. For completion then, here's how it looks with the oblate spheroid Earth, including the atmosphere code updates from @alexanderbock. Hits St Luis pretty well but I'm unsure about the timing

![image](https://user-images.githubusercontent.com/8808894/125091065-62c79b00-e0d0-11eb-8fbf-cdb6662c6cf1.png)

**Comment by alexanderbock** [comment]
Maybe related to the sun position to be hardcoded to (0,0,0)?  We have an issue for it, but I wasn't able to find it
